test/e2e/operator.go

```diff
@@ -891,7 +892,7 @@ var _ = Describe("ARO Operator - etchosts", func() {
 			}).WithContext(ctx).WithTimeout(DefaultEventuallyTimeout).Should(Succeed())
 		}
 
-		if !instance.Spec.OperatorFlags.GetSimpleBoolean(etchostsManaged) {
+		if !instance.Spec.OperatorFlags.GetSimpleBoolean(etchostsManaged) && instance.Spec.OperatorFlags.GetSimpleBoolean(forceReconciliation) {
```

Comment:
Can we reach this condition?
The default value of `etchostsManaged` is true. If we don't change the value elsewhere, we won't reach here.
Also it runs just once in each e2e test run, so only one of those conditions can be run, and others can't.

I think we should split this test cases into multiple test cases according to operator flags and test each behaviour in each test case not using conditional branches.

Reply:
I think I prefer the test as-is, because it is future-proof to our E2E test expanding out from the rigid structure it is in today (e.g. us performing permutation tests on various clusters with different features set in each one).

Reply:
Can't we change operator flags before running each test case?
I think we can because we can change cluster operator flags by running oc edit command.
If we can change the flags then, we can set them before each test, and we can test all scenarios.

Each test case should focus on one scenario, and this test case doesn't follow the principle.
Otherwise the purpose of the test cases will be unclear.
If we had to wait for the change in how to run E2E, we can rewrite this test case like below:

```go
It("must not reconcile machine configs when enabled == false", func(ctx context.Context) {
  if instance.Spec.OperatorFlags.GetSimpleBoolean(etchostsEnabled) {
    Skip("skipping because enabled == true")
  }
// ...
}
It("must reconcile machine configs when managed == true and ...", func(ctx context.Context) {
  if !instance.Spec.OperatorFlags.GetSimpleBoolean(etchostsManaged) || 
     !instance.Spec.OperatorFlags.GetSimpleBoolean(forceReconciliation) {
    Skip("skipping because managed == false || forceReconciliation == false")
  }
}
// ...
It("must remove machine configs when managed == false and ...", func(ctx context.Context) {
  if instance.Spec.OperatorFlags.GetSimpleBoolean(etchostsManaged) || 
     !instance.Spec.OperatorFlags.GetSimpleBoolean(forceReconciliation) {
      Skip("skipping because managed == true || forceReconciliation == false")
  }
}
```

Reply:
Which question is No? You mean we can't change operator flags before running each test case?

Also I want you to explain which part is wrong to get better understanding.